### PR TITLE
Issue #582: Add remark-lint to package.json test script

### PR DESCRIPTION
**Pull request by ChristianMurphy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,9 @@
   "scripts": {
     "pretest": "mvn clean package",
     "test": "karma start angularjs-portal-home/target/web/karma.conf.js  --single-run",
-  	"prestart": "mvn clean package",
-  	"start": "mvn jetty:run"
+    "posttest": "remark .",
```
Comment (thevoiceofzeke) [inline package.json]: Should include `--ignore-path .gitignore`

Comment (thevoiceofzeke) [inline package.json]: Also would prefer if it was its own script (i.e. "lint-md")

File: package.json [hunk 2]
```diff
@@ -35,6 +36,18 @@
...
+    "remark-cli": "^3.0.0",
+    "remark-lint": "^6.0.0",
+    "remark-preset-lint-consistent": "^2.0.0",
+    "remark-preset-lint-recommended": "^2.0.0",
+    "remark-validate-links": "^6.0.0",
     "requirejs": "^2.1.18"
+  },
+  "remarkConfig": {
+    "plugins": [
+      "preset-lint-recommended",
```
Comment (thevoiceofzeke) [inline package.json]: Default value for list-item-indent rule is creating a lot of irrelevant problems. Consider adding the following to the "plugins" array:

```json
[
    "remark-lint-list-item-indent",
    "mixed"
],
```

File: package.json (hunk 1 above)
Comment (ChristianMurphy) [inline package.json]: @thevoiceofzeke can do!
Should `posttest` become `lint-js` to match the other tasks?

Comment (ChristianMurphy) [inline package.json]: @thevoiceofzeke ignore setting has been configured, `lint-md` task has been added, eslint has been moved to `lint-js`.

File: package.json (hunk 2 above)
Comment (ChristianMurphy) [inline package.json]: @thevoiceofzeke Thanks!
That has been added.

**Pull request merged**